docs/man/mess.6
.\"  -*- nroff -*-
.\"
.\" mess.6
.\"
.\" Man page created from usage and source information:
.\" * commands: see src/emu/clifront.c clifront.h
.\" * options: core entries, see src/emu/emuopts.c emuopts.h
.\"            SDL-specific entries, see src/osd/sdl/sdlmain.c osdsdl.h
.\" Cesare Falco <c.falco@ubuntu.com>, March 2011
.\"
.\" Some text borrowed from the xmame 0.106 man page,
.\" done by Rene Herrmann <rene.herrmann@berlin.de>, September 2000
.\" and updated by Andrew Burton <burtona@gol.com>, July 2003
.\"
.\" Other info's taken from MESS Wiki as well:
.\" http://mess.redump.net
.\"
.\"
.TH MESS 6 2015-05-20 0.162 "MESS \- Multi Emulator Super System"
.\"
.\"
.\" NAME chapter
.SH NAME
MESS \- Multi Emulator Super System
.\"
.\"
.\" SYNOPSIS chapter
.SH SYNOPSIS
.B mess
.I system
.RI [ "media software" ]
.RI [ options ]
.\"
.\"
.\" DESCRIPTION chapter
.SH DESCRIPTION
MESS documents the hardware for a wide variety of (mostly vintage) computers,
video game consoles, and calculators through software emulation. As a nice
side effect, MESS allows software and games for these hardware platforms
to run on modern PCs.
.\"
.\"
.\" SOFTWARE chapter
.SH SOFTWARE
A \fIsoftware\fR entry may be specified to attach an image file to the
emulated system. In such a case, the appropriate \fImedia\fR switch
is also required:
.TP
.B \-cart, \-cart1
Cartrigde, used mainly by game consoles
.TP
.B \-cass
Cassette (tape), widely used by old home computers
.TP
.B \-flop, -flop1
Floppy disk
.TP
.B \-cdrom
CD-Rom
.\"
.\"
.\" OPTIONS chapter
.SH OPTIONS
.\"
.\" *******************************************************
.SS Core commands
.\" *******************************************************
.TP
.B \-help, \-?
Displays current MESS version and copyright notice.
.TP
.B \-validate, \-valid
Performs internal validation on every driver in MESS. Run this
before submitting changes to ensure that you haven't violated any of
the core system rules.
.\"
.\" *******************************************************
.SS Configuration commands
.\" *******************************************************
.TP
.B \-createconfig, \-cc
Creates the default \fBmess.ini\fR file in the current directory. All the
configuration options (not commands) described below can be permanently
changed by editing this configuration file.
.TP
.B \-showconfig, \-sc
Displays the current configuration settings.
.TP
.B \-showusage, \-su
Displays a summary of all the command line options. For options that
are not mentioned here, the short summary given by \fB\-showusage\fR
is usually sufficient.
.\"
.\" *******************************************************
.SS Frontend commands
.\" *******************************************************
All the list commands below write info to the screen.
If you wish to write the info to a textfile instead, use redirection.
For example,
.B mess \-listxml > ~/messlist.xml
writes the full list of supported systems to \fBmesslist.xml\fR in your
home directory.
.TP
.B \-listxml, \-lx \fR[\fIsystem\fR|\fIwildcard\fR]
List comprehensive details for all of the supported systems. The output
is quite long, so it is usually better to redirect this into a file.
The output is in XML format. By default all systems are listed; however,
you can limit this list by specifying a driver name or wildcard after
the \-listxml command.
.TP
.B \-listfull, \-ll \fR[\fIsystem\fR|\fIwildcard\fR]
Displays a list of system driver names and descriptions. By default all
systems are listed; however, you can limit this list by specifying a
driver name or wildcard after the \-listfull command.
.TP
.B \-listsource, \-ls \fR[\fIsystem\fR|\fIwildcard\fR]
Displays a list of drivers and the names of the source files their system
drivers live in. Useful for finding which driver a system runs on in
order to fix bugs. By default all systems are listed; however, you can
limit this list by specifying a driver name or wildcard after the
\-listsource command.
.TP
.B \-listclones, \-lc \fR[\fIsystem\fR|\fIwildcard\fR]
Displays a list of clones. By default all clones are listed; however,
you can limit this list by specifying a driver name or wildcard after
the \-listclones command.
.TP
.B \-listbrothers, \-lb \fR[\fIsystem\fR|\fIwildcard\fR]
Displays a list of "brothers" or other drivers from same sourcefile.
By default all systems are listed; however, you can limit this list by
specifying a driver name or wildcard after the \-listbrothers command.
.TP
.B \-listcrc
Displays a full list of CRCs of all ROM images referenced by all
drivers within MESS code.
.TP
.B \-listroms, \-lr \fIsystem
Displays a list of ROM images referenced by the specified system.
.TP
.B \-listsamples \fIsystem
Displays a list of samples referenced by the specified system.
.TP
.B \-verifyroms \fR[\fIsystem\fR|\fIwildcard\fR]
Checks for invalid or missing ROM images. By default all drivers that
have valid ZIP files or directories in the rom path are verified;
however, you can limit this list by specifying a driver name or
wildcard after the \-verifyroms command.
.TP
.B \-verifysamples \fR[\fIsystem\fR|\fIwildcard\fR]
Checks for invalid or missing samples. By default all drivers that
have valid ZIP files or directories in the samplepath are verified;
however, you can limit this list by specifying a driver name or
wildcard after the \-verifyroms command.
.TP
.B \-romident
Attempts to identify ROM files, if they are known to MESS, in the
specified .zip file or directory. This command can be used to try and
identify ROM sets taken from unknown boards. On exit, the errorlevel
is returned as one of the following:
.br
\fB0\fR  all files were identified
.br
\fB7\fR  all files were identified except for some "non\-ROM" files
.br
\fB8\fR  some files were identified
.br
\fB9\fR  no files were identified
.TP
.B \-listdevices, \-ld
Output the list of devices referenced by a given system or set of systems.
.TP
.B \-listslots, \-lslot
Output the list of available slots and slot devices for the system.
.TP
.B \-listmedia, \-lm
Output the list of available media for the system.
.TP
.B \-listsoftware, \-lsoft
Output the list of known software for the system.
.TP
.B \-verifysoftware, \-vsoft \fR[\fIgamename\fR|\fIwildcard\fR]
Checks for invalid or missing ROM images in your software lists.
By default all drivers that have valid ZIP files or directories
in the rompath are verified; however, you can limit this list by
specifying a specific driver name or wildcard.
.TP
.B \-getsoftlist, \-glist
Retrieve software list by name.
.TP
.B \-verifysoftlist, \-vlist \fR[\fIsoftwarelistname\fR]
Checks a specified software list for missing ROM images if files exist
for issued softwarelistname. By default, all drivers that have valid
ZIP files or directories in the rompath are verified; however, you can
limit this list by specifying a specific softwarelistname (without .XML).
.TP
.B \-listmidi, \-mlist
List available MIDI I/O devices.
.TP
.B \-listnetwork, \-nlist
List available network adapters.
.\"
.\" *******************************************************
.SS Configuration options
.\" *******************************************************
.TP
.B \-[no]readconfig, \-[no]rc
Enables or disables the reading of the config files. When enabled
(which is the default), MESS reads the following config files in order:
.br
1. \fBmess.ini\fR
.br
the main configuration file
.br
2. \fI[driver]\fB.ini\fR
.br
based on the source file name of the system driver
.br
3. \fBdebug.ini\fR, if the debugger is enabled
.br
4. \fBvector.ini\fR, for vector games only
.br
5. \fI[driver]\fB.ini\fR
.br
based on the source filename of the game driver
.br
6. \fI[parent]\fB.ini\fR
.br
for clones only, may be called recursively
.br
4. \fI[system]\fB.ini\fR
.br
note this sometimes resolves to the same of the source driver
.P
The settings in the later ini's override those in the earlier ini's.
The default is ON (\-readconfig).
.TP
.B \-[no]writeconfig, \-[no]wc
Write configuration to \fI[driver]\fR.ini on exit.
Default is OFF (\-nowriteconfig).
.\"
.\" *******************************************************
.SS Search path options
.\" *******************************************************
Be careful to use the path, directory and file options in
mess.ini ONLY. Otherwise, the outcome may be unpredictable and not
consistent across releases.
.TP
.B \-rompath, \-rp, \-biospath, \-bp \fIpathname
Specifies a list of paths within which to find ROM or hard disk images.
Multiple paths can be specified by separating them with semicolons.
The default is 'roms' (that is, a directory 'roms' in the same directory
as the MESS executable).
.TP
.B \-hashpath \fIpathname
Specifies a list of paths within which to search for software hash files.
Multiple paths can be specified by separating them with semicolons.
The default is 'hash' (that is, a directory 'hash' in the same directory
as the MESS executable).
.TP
.B \-samplepath, \-sp \fIpathname
Specifies a list of paths within which to find sample files. Multiple
paths can be specified by separating them with semicolons. The default
is 'samples' (that is, a directory 'samples' in the same directory as
the MESS executable).
.TP
.B \-artpath, \-artwork_directory \fIpathname
Specifies a list of paths within which to find artwork files. Multiple
paths can be specified by separating them with semicolons. The default
is 'artwork' (that is, a directory 'artwork' in the same directory as
the MESS executable).
.TP
.B \-ctrlrpath, \-ctrlr_directory \fIpathname
Specifies a list of paths within which to find controller\-specific
configuration files. Multiple paths can be specified by separating
them with semicolons. The default is 'ctrlr' (that is, a
directory 'ctrlr' in the same directory as the MESS executable).
.TP
.B \-inipath \fIpathname
Specifies a list of paths within which to find .ini files. Multiple
paths can be specified by separating them with semicolons. The default
is '/etc/mess'.
.TP
.B \-fontpath \fIpathname
Specifies a list of paths within which to find .bdf font files. Multiple
paths can be specified by separating them with semicolons. The default
is '.' (that is, search in the same directory as the MESS executable).
.TP
.B \-cheatpath \fIpathname
Specifies a list of paths within which to find cheat files. Multiple
paths can be specified by separating them with semicolons. The default
is 'cheat' (that is, a directory 'cheat' in the same directory as
the MESS executable).
.TP
.B \-crosshairpath \fIpathname
Specifies a list of paths within which to find crosshair files. Multiple
paths can be specified by separating them with semicolons. The default
is 'crosshair' (that is, a directory 'crosshair' in the same directory as
the MESS executable).  If the Crosshair is set to default in the menu,
MESS will look for system/cross#.png and then cross#.png in the
specified path, where # is the player number.  Failing that,
MESS will use built\-in default crosshairs.
.TP
.B \-pluginspath \fIpathname
Specifies a single path within which to find plugins. The default
is 'plugins' (that is, a directory 'plugins' in the same directory as
the MESS executable).
.TP
.B \-languagepath \fIpathname
Specifies a single path within which to find language files. The default
is 'language' (that is, a directory 'language' in the same directory as
the MESS executable).
.\"
.\" *******************************************************
.SS Output Directory Options
.\" *******************************************************
.TP
.B \-cfg_directory \fIpathname
Specifies a single directory where configuration files are stored.
Configuration files store user configurable settings that are read at
startup and written when MESS exits. The default is 'cfg' (that is,
a directory 'cfg' in the same directory as the MESS executable). If this
directory does not exist, it will be automatically created.
.TP
.B \-nvram_directory \fIpathname
Specifies a single directory where NVRAM files are stored. NVRAM files
store the contents of EEPROM and non\-volatile RAM (NVRAM) for systems
which used this type of hardware. This data is read at startup and
written when MESS exits. The default is 'nvram' (that is, a
directory 'nvram' in the same directory as the MESS executable). If this
directory does not exist, it will be automatically created.
.TP
.B \-input_directory \fIpathname
Specifies a single directory where input recording files are stored.
Input recordings are created via the \-record option and played back via
the \-playback option. The default is 'inp' (that is, a directory 'inp'
in the same directory as the MESS executable). If this directory
does not exist, it will be automatically created.
.TP
.B \-state_directory \fIpathname
Specifies a single directory where save state files are stored. Save
state files are read and written either upon user request, or when using
the \-autosave option. The default is 'sta' (that is, a directory 'sta'
in the same directory as the MESS executable). If this directory does
not exist, it will be automatically created.
.TP
.B \-snapshot_directory \fIpathname
Specifies a single directory where screen snapshots are stored, when
requested by the user. The default is 'snap' (that is, a directory 'snap'
in the same directory as the MESS executable). If this directory
does not exist, it will be automatically created.
.TP
.B \-diff_directory \fIpathname
Specifies a single directory where hard drive differencing files are
stored. Hard drive differencing files store any data that is written
back to a hard disk image, in order to preserve the original image. The
differencing files are created at startup when a system with a hard disk
image. The default is 'diff' (that is, a directory 'diff' in the same
directory as the MESS executable). If this directory does not exist,
it will be automatically created.
.TP
.B \-comment_directory \fIpathname
Specifies a single directory where debugger comment files are stored.
Debugger comment files are written by the debugger when comments are
added to the disassembly for a system. The default is 'comments' (that is,
a directory 'comments' in the same directory as the MESS executable).
If this directory does not exist, it will be automatically created.
.\"
.\" *******************************************************
.SS State/playback options
.\" *******************************************************
.TP
.B \-state \fIslot
Immediately after starting the specified system, will cause the save
state in the specified \fIslot\fP to be loaded.
.TP
.B \-[no]autosave
When enabled, automatically creates a save state file when exiting MESS
and automatically attempts to reload it when later starting MESS with
the same system. This only works for systems that have explicitly enabled
save state support in their driver. The default is OFF (\-noautosave).
.TP
.B \-playback, \-pb \fIfilename
Specifies a file from which to play back a series of system inputs. This
feature does not work reliably for all systems, but can be used to watch
a previously recorded system session from start to finish. In order to
make things consistent, you should only record and playback with all
configuration (.cfg), NVRAM (.nv), and memory card files deleted. The
default is NULL (no playback).
.TP
.B \-[no]exit_after_playback
If play back is enabled (see \-playback option) the program exits when
end of file is reached. Otherwise the machine will continue to run.
Default is OFF (\-noexit_after_playback).
.TP
.B \-record, \-rec \fIfilename
Specifies a file to record all input from a system session. This can be
used to record a system session for later playback. This feature does not
work reliably for all systems, but can be used to watch a previously
recorded system session from start to finish. In order to make things
consistent, you should only record and playback with all configuration
(.cfg), NVRAM (.nv), and memory card files deleted. The default is NULL
(no recording).
.TP
.B \-[no]record_timecode
Specify whether to create a timecode file. It contains a line with elapsed
times on each press of timecode shortcut key (default is F12).
This option works only when recording mode is enabled (\-record option).
The file is saved on inp folder.
Default is OFF (\-norecord_timecode).
.TP
.B \-mngwrite \fIfilename
Writes each video frame to the given file in MNG format, producing
an animation of the system session.
Note that \-mngwrite only writes video frames; it does not save any audio
data. Use \-wavwrite for that, and reassemble the audio/video using
offline tools. The default is NULL (no recording).
.TP
.B \-aviwrite \fIfilename
Stream video and sound data to the given file
in AVI format, producing an animation of the system session complete
with sound. The default is NULL (no recording).
.TP
.B \-wavwrite \fIfilename
Writes the final mixer output to the given file
in WAV format, producing an audio recording of the system session.
The default is NULL (no recording).
.TP
.B \-snapname \fIstring
Describes how MESS should name files for snapshots. \fIstring\fP
provides a template that is used to generate a filename. Three
simple substitutions are provided: the / character represents the
path separator on any target platform (even Windows); the literal \fB%g\fP
represents the driver name of the current system; and the literal \fB%i\fP
represents an incrementing index. If \fB%i\fP is omitted, then each
snapshot taken will overwrite the previous one; otherwise, MESS will
find the next empty value for \fB%i\fP and use that for a filename. The
default is \fB%g/%i\fP, which creates a separate folder for each system,
and names the snapshots under it starting with 0000 and increasing
from there. In addition to the above, for drivers using different
media, like carts or floppy disks, you can also use the
\fB%d\fP\fI_[media]\fP indicator. Replace \fI[media]\fP with the media
switch you want to use. A few examples: if you use
.B mess nes \-cart robby \-snapname %g/%d_cart
snapshots will be saved as \fBsnaps/nes/robby.png\fP; if you use
.B mess c64 \-flop1 robby \-snapname %g/%d_flop1/%i
snapshots will be saved as \fBsnaps/c64/robby/0000.png\fP.
.TP
.B \-snapsize \fIwidth\fRx\fIheight
Hard\-codes the size for snapshots and movie recording. By default,
MESS will create snapshots at the system's current resolution in raw
pixels, and will create movies at the system's starting resolution in
raw pixels. If you specify this option, then MESS will create both
snapshots and movies at the size specified, and will bilinear filter
the result. Note that this size does not automatically rotate if the
system is vertically oriented. The default is 'auto'.
.TP
.B \-snapview internal\fR|\fBauto\fR|\fIviewname
Specifies the view to use when rendering snapshots and movies. By
default, both use a special 'internal' view, which renders a separate
snapshot per screen or renders movies only of the first screen. By
specifying this option, you can override this default behavior and
select a single view that will apply to all snapshots and movies.
Note that \fIviewname\fP does not need to be a perfect match; rather, it
will select the first view whose name matches all the characters
specified by \fIviewname\fP. For example, \-snapview native will match the
"Native (15:14)" view even though it is not a perfect match.
\fIviewname\fP can also be 'auto', which selects the first view with all
screens present. The default value is 'internal'.
.TP
.B \-[no]snapbilinear
Specify whether snapshot/movie should have bilinear filtering applied.
Default is ON (\-snapbilinear).
.TP
.B \-statename \fIstring
Describes how MESS should store save state files, relative to the
state_directory path. \fIstring\fP provides a template that
is used to generate a relative path. Two simple substitutions are
provided: the / character represents the path separator on any target
platform (even Windows); the literal \fB%g\fP represents the driver name of
the current game. The default is '%g', which creates a separate folder
for each game. In addition to the above, for drivers using different
media, like carts or floppy disks, you can also use the
\fB%d_\fP\fI[media]\fP indicator. Replace \fI[media]\fP with the media
switch you want to use. A few examples: if you use
.B mess nes \-cart robby \-statename %g/%d_cart
save states will be stored inside \fBsta/nes/robby/\fP; if you use
.B mess c64 \-flop1 robby \-statename %g/%d_flop1
save states will be stored inside \fBsta/c64/robby/\fP.
.TP
.B \-[no]burnin
Tracks brightness of the screen during play and at the end of
emulation generates a PNG that can be used to simulate burn\-in
effects on other systems. The resulting PNG is created such that the
least\-used areas of the screen are fully white (since burned\-in areas
are darker, all other areas of the screen must be lightened a touch).
The intention is that this PNG can be loaded via an artwork file with
a low alpha (e.g, 0.1\-0.2 seems to work well) and blended over the
entire screen. The PNG files are saved in the snap directory under
the system/burnin\-<screen.name>.png. The default is OFF (\-noburnin).
.\"
.\" *******************************************************
.SS Performance options
.\" *******************************************************
.TP
.B \-[no]autoframeskip, \-[no]afs
Automatically determines the frameskip level while you're playing the
system, adjusting it constantly in a frantic attempt to keep the system
running at full speed. Turning this on overrides the value you have set
for \-frameskip below. The default is OFF (\-noautoframeskip).
.TP
.B \-frameskip, \-fs \fIvalue
Specifies the frameskip value (autoframeskip must be disabled). This is the
number of frames out of every 12 to drop when running. For example, if you
say \-frameskip 2, then MESS will display 10 out of every 12 frames. By
skipping those frames, you may be able to get full speed in a system that
requires more horsepower than your computer has. The default value is 0,
which skips no frames.
.TP
.B \-seconds_to_run, \-str \fIvalue
This option can be used for benchmarking and automated testing. It tells
MESS to stop execution after a fixed number of seconds. By combining
this with a fixed set of other command line options, you can set up a
consistent environment for benchmarking MESS performance. In addition,
upon exit, the \-str option will write a screenshot called final.png
to the system's snapshot directory.
.TP
.B \-[no]throttle
Configures the default thottling setting. When throttling is on, MESS
attempts to keep the system running at the system's intended speed. When
throttling is off, MESS runs the system as fast as it can. Note that the
fastest speed is more often than not limited by your graphics card,
especially for older systems. The default is ON (\-throttle).
.TP
.B \-[no]sleep
Allows MESS to give time back to the system when running with \-throttle.
This allows other programs to have some CPU time, assuming that the
system isn't taxing 100% of your CPU resources. This option can potentially
cause hiccups in performance if other demanding programs are running.
The default is ON (\-sleep).
.TP
.B \-speed \fIvalue
Controls the speed of gameplay, relative to realtime; smaller numbers are
slower. Default is 1.00.
.TP
.B \-[no]refreshspeed, \-[no]rs
Automatically adjusts the \fB\-speed\fR parameter to keep the effective refresh
rate below that of the lowest screen refresh rate.
Default is OFF (\-norefreshspeed).
.\"
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.\" OS specific
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.TP
.B \-numprocessors, \-np \fIvalue
Set number of processors; this overrides the number the system reports.
.TP
.B \-bench \fIvalue
Benchmark for the given number of emulated seconds;
implies \-video none \-sound none \-nothrottle.
.\"
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.\" SDL specific
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.TP
.B \-[no]sdlvideofps
Show SDL video performance. Default is OFF (\-nosdlvideofps).
.\"
.\" *******************************************************
.SS Rotation options
.\" *******************************************************
.TP
.B \-[no]rotate
Rotate the system to match its normal state (horizontal/vertical). This
ensures that both vertically and horizontally oriented systems show up
correctly without the need to rotate your monitor. If you want to keep
the system displaying 'raw' on the screen the way the actual machine would
have, turn this option off. The default is ON (\-rotate).
.TP
.B \-[no]ror \-[no]rol
Rotate the system screen to the right (clockwise) or left
(counter\-clockwise) relative to either its normal state (if \-rotate
is specified) or its native state (if \-norotate is specified).
Default for both is OFF (\-noror \-norol).
.TP
.B \-[no]autoror \-[no]autorol
These options are designed for use with pivoting screens that only
pivot in a single direction. If your screen only pivots clockwise,
use \-autorol to ensure that the system will fill the screen either
horizontally or vertically in one of the directions you can handle.
If your screen only pivots counter\-clockwise, use \-autoror.
Default for both is OFF (\-noautoror \-noautorol).
.TP
.B \-[no]flipx \-[no]flipy
Flip (mirror) the system screen either horizontally (\-flipx) or
vertically (\-flipy). The flips are applied after the \-rotate and
\-ror/\-rol options are applied.
Default for both is OFF (\-noflipx \-noflipy).
.\"
.\" *******************************************************
.SS Artwork options
.\" *******************************************************
.TP
.B \-[no]artwork_crop, \-[no]artcrop
Enable cropping of artwork to the system screen area only. This
option can also be controlled via the Video Options menu in the user
interface. The default is OFF (\-noartwork_crop).
.TP
.B \-[no]use_backdrops, \-[no]backdrop
Enables/disables the display of backdrops. The default is ON
(\-use_backdrops).
.TP
.B \-[no]use_overlays, \-[no]overlay
Enables/disables the display of overlays. The default is ON
(\-use_overlays).
.TP
.B \-[no]use_bezels, \-[no]bezel
Enables/disables the display of bezels. The default is ON
(\-use_bezels).
.TP
.B \-[no]use_cpanels, \-[no]cpanel
Enables/disables the display of cpanels. The default is ON
(\-use_bezels).
.TP
.B \-[no]use_marquees, \-[no]marquee
Enables/disables the display of marquees. The default is ON
(\-use_bezels).
.\"
.\" *******************************************************
.SS Screen options
.\" *******************************************************
.TP
.B \-brightness \fIvalue
Controls the default brightness, or black level, of the system screens.
This option does not affect the artwork or other parts of the display.
Using the MESS UI, you can individually set the brightness for each system
screen; this option controls the initial value for all visible system
screens. The standard value is 1.0. Selecting lower values (down to 0.1)
will produce a darkened display, while selecting higher values (up to
2.0) will give a brighter display. The default is 1.0.
.TP
.B \-contrast \fIvalue
Controls the contrast, or white level, of the system screens. This option
does not affect the artwork or other parts of the display. Using the
MESS UI, you can individually set the contrast for each system screen;
this option controls the initial value for all visible system screens. The
standard value is 1.0. Selecting lower values (down to 0.1) will produce
a dimmer display, while selecting higher values (up to 2.0) will
give a more saturated display. The default is 1.0.
.TP
.B \-gamma \fIvalue
Controls the gamma, which produces a potentially nonlinear black to
white ramp, for the system screens. This option does not affect the
artwork or other parts of the display. Using the MESS UI, you can
individually set the gamma for each system screen; this option controls
the initial value for all visible system screens. The standard value is
1.0, which gives a linear ramp from black to white. Selecting lower
values (down to 0.1) will increase the nonlinearity toward black,
while selecting higher values (up to 3.0) will push the nonlinearity
toward white. The default is 1.0.
.TP
.B \-pause_brightness \fIvalue
This controls the brightness level when MESS is paused. The default
value is 0.65.
.TP
.B \-effect\fR [\fInone\fR|\fIfilename[.png]\fR]
Name of a PNG file to use for visual effects, or 'none'. Default is 'none'.
.\"
.\" *******************************************************
.SS Vector rendering options
.\" *******************************************************
.TP
.B \-beam_width_min \fIvalue
.TP
.B \-beam_width_max \fIvalue
Sets the minimum and maximum width of the vectors. This is a scaling factor
against the standard vector width, which is interpolated between minimum and
maximum according to the beam's intensity. A value of 1.0 will keep the
default vector line width. Smaller values will reduce the width, and larger
values will increase the width. The default is 1.0.
.TP
.B \-beam_intensity_weight \fIvalue
Applies an exponential weight to the minimum and maximum beam width. For
positive values the interpolated scaling factor will affect lines with higher
intensity more than lines with lower intensity. The default is 0.0.
.TP
.B \-flicker \fIvalue
Simulates a vector "flicker" effect, similar to a vector monitor that
needs adjustment. This option requires a float argument in the range of
0.00\-100.00 (0=none, 100=maximum). The default is 0.
.\"
.\" *******************************************************
.SS Video options
.\" *******************************************************
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.\" SDL specific
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.TP
.B \-video\fR [\fIsoft\fR|\fIopengl\fR|\fIbgfx\fR|\fInone\fR]
Specifies which video subsystem to use for drawing:
.br
\fBsoft\fR  uses software rendering, which is slower but more compatible.
.br
\fBopengl\fR  uses OpenGL and your graphics accelerator to speed up many
aspects of drawing MAME including compositing artwork, overlays, and
bezels, as well as stretching the image to fit your screen.
.br
\fBbgfx\fR  uses the bgfx renderer which supports portable shaders and
multiple graphics APIs including OpenGL, OpenGL ES and DirectX.
.br
\fBnone\fR  does no drawing and is intended for CPU benchmarking.
.br
Default is 'soft' on Linux or 'opengl' on macOS.
.TP
.B \-numscreens \fR[\fI1-4\fR]
Number of screens to create; usually, you want just one. Default is '1'.
.TP
.B \-[no]window, \-[no]w
Run MESS in either full screen or a window. This is a fully\-featured window
mode where the window resizes as necessary to track what the system does.
And you can resize it yourself with your OS's standard window controls.
The default is OFF (\-nowindow).
.TP
.B \-[no]maximize, \-[no]max
Controls initial window size in windowed mode. If it is set on, the
window will initially be set to the maximum supported size when you
start MESS. If it is turned off, the window will start out at the
smallest supported size. This option only has an effect when the
\-window option is used. The default is ON (\-maximize).
.TP
.B \-[no]keepaspect, \-[no]ka
Forces the correct aspect ratio. This means when you're resizing the window
in windowed mode the actual system image will resize in discrete steps to
maintain the proper shape of the system graphics. If you turn this off you can
resize the window to anything you like and get funny squishing and stretching.
The same applies for full\-screen. Default is ON (\-keepaspect).
.TP
.B \-[no]unevenstretch, \-[no]ues
Allow non\-integer stretch factors. Video purists should stay far, far away
from this option, while everyone else will be happy to know that it lets you
fill the screen properly in full\-screen mode. Default is ON (\-unevenstretch).
.TP
.B \-[no]unevenstretchx, \-[no]uesx
Act as \-[no]unevenstretch on horizontal basis only.
.TP
.B \-[no]intoverscan, \-[no]ios
Allow overscan on integer scaled targets.
.TP
.B \-intscalex, \-sx
Set horizontal integer scale factor.
.TP
.B \-intscaley, \-sy
Set vertical integer scale factor.
.TP
.B \-[no]centerh
Center horizontally within the view area. Default is ON (\-centerh).
.TP
.B \-[no]centerv
Center vertically within the view area. Default is ON (\-centerv).
.TP
.B \-[no]waitvsync, \-[no]vs
Enable waiting for the start of VBLANK before flipping screens;
reduces tearing effects. Default is OFF (\-nowaitvsync).
.TP
.B \-[no]syncrefresh, \-[no]srf
Enable using the start of VBLANK for throttling instead of the game time.
Default is OFF (\-nosyncrefresh).
.\"
.\" *******************************************************
.SS Software video rendering subsystem options
.\" *******************************************************
.\" +++++++++++++++++++++++++++++++++++++++++++++++++++++++
.\" SDL specific

src/mame/mess.c
// license:BSD-3-Clause
// copyright-holders:Aaron Giles
/***************************************************************************

    mess.c

    Specific (per target) constants

****************************************************************************/

#include "emu.h"

#define APPNAME                 "MESS"
#define APPNAME_LOWER           "mess"
#define CONFIGNAME              "mess"
#define APPLONGNAME             "M.E.S.S."
#define FULLLONGNAME            "Multi Emulator Super System"
#define CAPGAMENOUN             "SYSTEM"
#define CAPSTARTGAMENOUN        "System"
#define GAMENOUN                "system"
#define GAMESNOUN               "systems"
#define COPYRIGHT               "Copyright Nicola Salmoria\nand the MAME team\nhttp://mamedev.org"
#define COPYRIGHT_INFO          "Copyright Nicola Salmoria and the MAME team"
#define DISCLAIMER              "MESS is an emulator: it reproduces, more or less faithfully, the behaviour of\n"\
								"several computer and console systems. But hardware is useless without software\n" \
								"so a file dump of the ROM, cartridges, discs, and cassettes which run on that\n" \
								"hardware is required. Such files, like any other commercial software, are\n" \
								"copyrighted material and it is therefore illegal to use them if you don't own\n" \
								"the original media from which the files are derived. Needless to say, these\n" \
								"files are not distributed together with MESS. Distribution of MESS together\n" \
								"with these files is a violation of copyright law and should be promptly\n" \
								"reported to the authors so that appropriate legal action can be taken.\n"
#define USAGE                   "Usage:  %s [%s] [media] [software] [options]"
#define XML_ROOT                "mess"
#define XML_TOP                 "machine"
#define STATE_MAGIC_NUM         "MESSSAVE"

const char * emulator_info::get_appname() { return APPNAME;}
const char * emulator_info::get_appname_lower() { return APPNAME_LOWER;}
const char * emulator_info::get_configname() { return CONFIGNAME;}
const char * emulator_info::get_applongname() { return APPLONGNAME;}
const char * emulator_info::get_fulllongname() { return FULLLONGNAME;}
const char * emulator_info::get_capgamenoun() { return CAPGAMENOUN;}
const char * emulator_info::get_capstartgamenoun() { return CAPSTARTGAMENOUN;}
const char * emulator_info::get_gamenoun() { return GAMENOUN;}
const char * emulator_info::get_gamesnoun() { return GAMESNOUN;}
const char * emulator_info::get_copyright() { return COPYRIGHT;}
const char * emulator_info::get_copyright_info() { return COPYRIGHT_INFO;}
const char * emulator_info::get_disclaimer() { return DISCLAIMER;}
const char * emulator_info::get_usage() { return USAGE;}
const char * emulator_info::get_xml_root() { return XML_ROOT;}
const char * emulator_info::get_xml_top() { return XML_TOP;}
const char * emulator_info::get_state_magic_num() { return STATE_MAGIC_NUM;}
void emulator_info::printf_usage(const char *par1, const char *par2) { osd_printf_info(USAGE, par1, par2); }
